lib/gears/surface.lua

```diff
@@ -283,6 +284,82 @@ function surface.widget_to_surface(widget, width, height, format)
     return img, run_in_hierarchy(widget, cr, width, height)
 end
 
+--- Crop a surface on its edges
+-- @tparam[opt=nil] table args
+-- @tparam int args.left Left cutoff, cannot be negative
+-- @tparam int args.right Right cutoff, cannot be negative
+-- @tparam int args.top Top cutoff, cannot be negative
+-- @tparam int args.bottom Bottom cutoff, cannot be negative
+-- @tparam number args.ratio ratio to crop the image to. If edge cutoffs and
+-- ratio are given, the edge cutoffs are computed first. Using ratio will crop
+-- the center out of an image, similar to what "zoomed-fill" does in wallpaper
+-- setter programs. Cannot be negative
+-- @tparam[opt=nil] surface args.surface the surface to crop
+-- @return the cropped surface
+-- @staticfct crop_surface
+function surface.crop_surface(args)
+    args = args or {}
+
+    if not args.surface then
+        gdebug.print_error("No surface to crop_surface supplied")
+        return nil
+    end
+
+    local surf = args.surface
+    local target_ratio = args.ratio
+
+    local w, h = surface.get_size(surf)
+    local offset_w, offset_h =  0, 0
+
+    if (args.top or args.right or args.bottom or args.left) then
+        local left = args.left or 0
+        local right = args.right or 0
+        local top = args.top or 0
+        local bottom = args.bottom or 0
+
+        if (top < 0 or right < 0 or bottom < 0 or left < 0) then
+            gdebug.print_error("negative offsets are not supported for crop_surface")
```

Comment:
i think error() would make more sense here and in most of gdebug.print_error() occurences which i see around